flixel/addons/display/FlxShaderMaskCamera.hx

```diff
@@ -0,0 +1,288 @@
+		
+		// Apply the provided shader using a ShaderFilter
+		shaderFilter = new ShaderFilter(effectShader);
+		_shadedScrollRect.filters = [shaderFilter];
+	}
+
+	override function update(elapsed:Float):Void
+	{
+		super.update(elapsed);
+		_maskGroup.update(elapsed);
+	}
+
+	override function render():Void
+	{
+		// clear our duplicate canvas 
+		shaderCanvas.graphics.clear();
+		super.fill(bgColor.to24Bit(), useBgAlphaBlending, bgColor.alphaFloat, shaderCanvas.graphics);
+		// iterate over draw items, but draw them to both canvases
+		var currItem:FlxDrawBaseItem<Dynamic> = _headOfDrawStack;
+		var oldCanvas:Sprite = this.canvas;
+		while (currItem != null)
+		{
+			// render to main canvas
+			currItem.render(this);
+			// render to shader canvas
+			this.canvas = this.shaderCanvas;
+			currItem.render(this);
+			// revert canvas
+			this.canvas = oldCanvas;
+			currItem = currItem.next;
+		}
+		// reset these to avoid re-drawing all other draw items to the mask canvas
+		// this is safe, since draw items are disposed by iterating over _headTiles and _headTriangles
+		_currentDrawItem = null;
+		_headOfDrawStack = null;
+		// populate the draw stack with mask items
+		this._maskGroup.draw();
+		// render draw stack to mask canvas
+		maskCanvas.graphics.clear();
+		this.canvas = maskCanvas;
+		currItem = _headOfDrawStack;
+		while (currItem != null)
+		{
+			currItem.render(this);
+			currItem = currItem.next;
+		}
+		this.canvas = oldCanvas;
+	}
+
+	/**
+	 * Add objects and groups to the mask.
+	 * The effect shader is applied to the area covered by objects added in this way.
+	 * The camera will `update()` objects added here, so you probably want to avoid
+	 * adding them to your scene as well.
+	 * 
+	 * @return  The same `FlxBasic` object that was passed in.
+	 */
+	public function addMaskObject(object:FlxBasic):FlxBasic {
+		object.camera = this;
+		return _maskGroup.add(object);
+	}
+
+	/**
+	 * Remove objects and groups from the mask.
+	 * 
+	 * @return  The removed object.
+	 */
+	public function removeMaskObject(object:FlxBasic):FlxBasic {
+		return _maskGroup.remove(object);
+	}
+
+	override function destroy():Void
+	{
+		FlxDestroyUtil.removeChild(flashSprite, _shadedScrollRect);
+		FlxDestroyUtil.removeChild(_shadedScrollRect, shaderCanvas);
+		if (shaderCanvas != null)
+		{
+			for (i in 0...shaderCanvas.numChildren)
+			{
+				shaderCanvas.removeChildAt(0);
+			}
+			shaderCanvas = null;
+		}
+		if (maskCanvas != null)
+		{
+			for (i in 0...maskCanvas.numChildren)
+			{
+				maskCanvas.removeChildAt(0);
+			}
+			maskCanvas = null;
+		}
+
+		if (_maskGroup != null) 
+		{
+			_maskGroup.destroy();
+			_maskGroup = null;
+		}
+
+		_shadedScrollRect = null;
+		shaderFilter = null;
+
+		super.destroy();
+	}
+
+	// Apply position/size changes to our duplicate and mask canvases
+	override function updateInternalSpritePositions():Void
+	{
+		super.updateInternalSpritePositions();
+		for (iCanvas in [shaderCanvas, maskCanvas])
```

Comment:
What's an `iCanvas`, is that a new Apple product I missed? :D

Reply:
Using `canvas` would just shadow the class variable inside the scope of the `for` loop.

Reply:
Well, as a general rule, it may be helpful, but here the loop is tiny (~6 lines), it is unlikely the reader would lose track of the iteration variable declared right there.

Reply:
The strange variable name definitely causes more confusion.